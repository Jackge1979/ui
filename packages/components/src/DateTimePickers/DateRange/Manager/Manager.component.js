import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { DateRangeContext } from '../Context';
import { extractParts, extractPartsFromTextInputRange } from '../date-range-extraction';
import { extractFromDate } from '../../Date/date-extraction';
import { START_DATE, END_DATE } from '../constants';

function ContextualManager(props) {
	function getOptions() {
		return {
			dateFormat: props.dateFormat,
		};
	}
	const initialState = extractParts(props.startDate, props.endDate, getOptions());
	const [state, setState] = useState(initialState);

	useEffect(() => {
		if (props.startDate !== state.startDate || props.endDate !== state.endDate) {
			const parts = extractParts(props.startDate, props.endDate, getOptions());
			setState(parts);
		}
	}, [props.startDate, props.endDate]);

	function onDatesChange(event, nextState) {
		if (props.onChange) {
			const payload = {
				startDate: nextState.startDate,
				endDate: nextState.endDate,
				errors: nextState.errors,
				errorMessage: nextState.errorMessage,
				field: state.focusedInput,
				origin: 'RANGE_PICKER',
			};
			props.onChange(event, payload, nextState.focusedInput);
		}
	}

	function onFocusChange(event, focusedInput) {
		setState({
			...state,
			focusedInput,
		});
	}

	function onSelectDate(event, { date }) {
		const parts = extractFromDate(date, getOptions());
		let dateParts;
		if (state.focusedInput === START_DATE) {
			dateParts = {
				startDate: parts.date,
				startDateTextInput: parts.textInput,
				focusedInput: END_DATE,
			};
		} else if (state.focusedInput === END_DATE) {
			dateParts = {
				endDate: parts.date,
				endDateTextInput: parts.textInput,
				focusedInput: state.startDate ? null : START_DATE,
			};
		}
		const nextState = {
			...state,
			...dateParts,
			errors: parts.errors,
			errorMessage: parts.errorMessage,
		};
		setState(nextState);
		onDatesChange(event, nextState);
	}

	function onInputChange(event) {
		const textInput = event.target.value;
		const parts = extractPartsFromTextInputRange(textInput, state.focusedInput, getOptions());
		const nextState = { ...state, ...parts };
		setState(nextState);
		onDatesChange(event, nextState);
	}

	return (
		<DateRangeContext.Provider
			value={{
				startDate: {
					value: state.startDate,
					textInput: state.startDateTextInput,
				},
				endDate: {
					value: state.endDate,
					textInput: state.endDateTextInput,
				},
				inputManagement: {
					onChange: onInputChange,
					onFocus: onFocusChange,
					focusedInput: state.focusedInput,
				},
				pickerManagement: {
					onSubmit: onSelectDate,
				},
			}}
		>
			{props.children}
		</DateRangeContext.Provider>
	);
}
ContextualManager.defaultProps = {
	dateFormat: 'YYYY-MM-DD',
};
ContextualManager.displayName = 'DateRange.Manager';
ContextualManager.propTypes = {
	children: PropTypes.element,
	startDate: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.number, PropTypes.string]),
	endDate: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.number, PropTypes.string]),
	onDatesChange: PropTypes.func.isRequired,
};

export default ContextualManager;
