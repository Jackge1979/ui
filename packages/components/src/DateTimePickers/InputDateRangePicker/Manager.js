import React, { useState, useRef, useEffect } from 'react';

import { DateRangeContext } from '../DateRange/Context';

function Manager(props) {
	const [state, setState] = useState({
		focusedInput: null,
		startDate: props.startDate,
		endDate: props.endDate,
	});
	let startRef = useRef(null);
	let endRef = useRef(null);

	useEffect(() => {
		if (state.focusedInput === 'startDate' && startRef) {
			startRef.focus();
		} else if (state.focusedInput === 'endDate' && endRef) {
			endRef.focus();
		}
	});
	function onChange(event, payload) {
		const nextState = {};
		if (state.focusedInput === 'startDate') {
			nextState.startDate = payload.date;
			nextState.focusedInput = 'endDate';
		} else if (state.focusedInput === 'endDate') {
			nextState.endDate = payload.date;
		}
		setState({ ...state, ...nextState });
		if (props.onChange) {
			props.onChange(event, { ...state, ...nextState });
		}
	}
	function onFocus(event, field) {
		setState({ ...state, focusedInput: field });
	}
	return (
		<DateRangeContext.Provider
			value={{
				startDate: state.startDate,
				endDate: state.endDate,
				focusedInput: state.focusedInput,
				setStartRef: ref => (startRef = ref),
				setEndRef: ref => (endRef = ref),
				onChange,
				onFocus,
			}}
		>
			{props.children}
		</DateRangeContext.Provider>
	);
}

export default Manager;
