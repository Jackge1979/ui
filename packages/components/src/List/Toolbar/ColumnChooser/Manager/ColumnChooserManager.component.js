import React from 'react';
import PropTypes from 'prop-types';
// import cloneDeep from 'lodash/cloneDeep';

const createColumnChooserData = column => {
	return {
		hidden: column.hidden,
		label: column.label,
		locked: column.locked,
		order: column.order,
	};
};

export default class ColumnChooserManager extends React.Component {
	static propTypes = {
		id: PropTypes.string,
		children: PropTypes.func.isRequired,
		columns: PropTypes.array.isRequired,
		handlerColumnChooser: PropTypes.func.isRequired,
	};

	state = {
		// columns: this.props.columns,
		// editedColumns: cloneDeep(this.props.columns),
		editedColumns: this.props.columns && this.props.columns.map(createColumnChooserData),
	};

	onClickModify = event => {
		this.setState(
			prevState => {
				return {
					editedColumns: prevState.editedColumns,
				};
			},
			() => this.props.handlerColumnChooser(event, this.state.editedColumns),
		);
	};

	onChangeVisibilityColumn = (index, hidden) => {
		this.setState(prevState => {
			const editedColumns = prevState.editedColumns;
			editedColumns[index].hidden = hidden;
			return { editedColumns };
		});
	};

	onChangeOrderColumn = (event, index) => {
		const value = event.target.value;
		this.setState(prevState => {
			const editedColumns = prevState.editedColumns;
			editedColumns[index].order = value;
			return { editedColumns };
		});
	};

	getColumns() {
		return this.state.editedColumns;
	}

	reset = () => {
		this.props.handlerColumnChooser(event, this.props.columns);
	};

	render() {
		const { children } = this.props;
		if (typeof children !== 'function') {
			throw new Error('ColumnChooser manager needs to have children as function');
		}
		return children({
			columnChooserId: this.props.id,
			columns: this.getColumns(),
			onClickModify: this.onClickModify,
			onChangeVisibilityColumn: this.onChangeVisibilityColumn,
			onChangeOrderColumn: this.onChangeOrderColumn,
			reset: this.reset,
		});
	}
}
