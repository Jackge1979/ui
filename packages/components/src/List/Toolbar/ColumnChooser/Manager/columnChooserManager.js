import { useState } from 'react';

export function compareOrder(a, b) {
	if (!Number.isInteger(a.order) && !Number.isInteger(b.order)) {
		return 0;
	}
	if (Number.isInteger(a.order) && !Number.isInteger(b.order)) {
		return -1;
	}
	if (!Number.isInteger(a.order) && Number.isInteger(b.order)) {
		return 1;
	}
	return a.order - b.order;
}

function useColumnChooserManager(columns, submitChanges) {
	const [editedColumns, setEditedColumns] = useState({ editedColumns: columns });

	function sortEditedColumns() {
		setEditedColumns(prevState => {
			return {
				editedColumns: prevState.editedColumns.sort(compareOrder).map((col, index) => {
					return { ...col, order: index + 1 };
				}),
			};
		});
	}

	function submitColumns(event) {
		submitChanges(event, editedColumns);
	}

	function changeColumnVisibility(index) {
		return function changeVisiblity(value) {
			setEditedColumns(prevState => {
				const inEditColumns = prevState.editedColumns;
				inEditColumns[index].hidden = value;
				return { editedColumns: inEditColumns };
			});
		};
	}

	function changeColumnOrder(index) {
		return function changeOrder(value) {
			setEditedColumns(prevState => {
				const inEditColumns = prevState.editedColumns;
				inEditColumns[index].order = value;
				return { editedColumns: inEditColumns };
			});
		};
	}

	function onDragAndDrop() {
		console.log('onDragAndDrop');
		sortEditedColumns();
		/*
		 = order / + 1 order
		*/
		// const dropInIndex = editedColumns.findIndex(column => column.label === dropIn.label);
		// const dropOnIndex = editedColumns.findIndex(column => column.label === dropOn.label);
		// const arrayTmp = editedColumns.editedColumns;
	}

	return {
		editedColumns: editedColumns.editedColumns,
		submitColumns,
		changeColumnOrder,
		changeColumnVisibility,
		onDragAndDrop,
	};
}

export { useColumnChooserManager };
