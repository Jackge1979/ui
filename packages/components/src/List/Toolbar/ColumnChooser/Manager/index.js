import ColumnChooserManager from './ColumnChooserManager.component';

export default ColumnChooserManager;
