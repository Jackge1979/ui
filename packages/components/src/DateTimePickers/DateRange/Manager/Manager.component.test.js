import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import cases from 'jest-in-case';

import Manager from './Manager.component';
import { DateRangeContext } from '../Context';

const DEFAULT_ID = 'DEFAULT_ID';

function DateRangeConsumerDiv() {
	return <div />;
}
// eslint-disable-next-line react/prop-types
function DateRangeConsumer() {
	return (
		<DateRangeContext.Consumer>
			{contextValue => <DateRangeConsumerDiv {...contextValue} />}
		</DateRangeContext.Consumer>
	);
}

describe('DateRange.Manager', () => {
	it('should render its children', () => {
		// when
		const wrapper = shallow(
			<Manager id={DEFAULT_ID} startDate={new Date(2017, 3, 4)} endDate={new Date(2017, 3, 10)}>
				<DateRangeConsumer />
			</Manager>,
		);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	describe('state management', () => {
		cases(
			'initial state',
			({ startDate, endDate, expectedStartDate, expectedEndDate }) => {
				// when
				const wrapper = mount(
					<Manager id={DEFAULT_ID} startDate={startDate} endDate={endDate}>
						<DateRangeConsumer />
					</Manager>,
				);

				// then
				const contextValue = wrapper.find('DateRangeConsumerDiv').props();
				expect(contextValue.startDate).toEqual(expectedStartDate);
				expect(contextValue.endDate).toEqual(expectedEndDate);
			},
			[
				{
					name: 'should init default state',
					startDate: undefined,
					endDate: undefined,
					expectedStartDate: {
						value: undefined,
						textInput: '',
					},
					expectedEndDate: {
						value: undefined,
						textInput: '',
					},
				},
				{
					name: 'should init default state from props invalid date',
					startDate: undefined,
					endDate: undefined,
					expectedStartDate: {
						value: undefined,
						textInput: '',
					},
					expectedEndDate: {
						value: undefined,
						textInput: '',
					},
				},
				{
					name: 'should init state from date object',
					startDate: new Date(2019, 9, 1),
					endDate: new Date(2019, 9, 10),
					expectedStartDate: {
						value: new Date(2019, 9, 1),
						textInput: '2019-10-01',
					},
					expectedEndDate: {
						value: new Date(2019, 9, 10),
						textInput: '2019-10-10',
					},
				},
				{
					name: 'should init state from datetime string',
					startDate: '2019-10-01',
					endDate: '2019-10-10',
					expectedStartDate: {
						value: new Date(2019, 9, 1),
						textInput: '2019-10-01',
					},
					expectedEndDate: {
						value: new Date(2019, 9, 10),
						textInput: '2019-10-10',
					},
				},
			],
		);

		// cases(
		// 	'props update should update state',
		// 	({ initialDate, newDate, expectedDate, expectedTime, useSeconds }) => {
		// 		// given
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} value={initialDate} useSeconds={useSeconds}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);

		// 		// when
		// 		act(() => {
		// 			wrapper.setProps({
		// 				value: newDate,
		// 			});
		// 		});
		// 		wrapper.update();

		// 		// then
		// 		const contextValue = wrapper.find('DateRangeConsumerDiv').props();
		// 		expect(contextValue.date).toEqual(expectedDate);
		// 		expect(contextValue.time).toEqual(expectedTime);
		// 	},
		// 	[
		// 		{
		// 			name: 'from undefined props(Date type)',
		// 			initialDate: new Date(),
		// 			newDate: undefined,
		// 			expectedDate: undefined,
		// 			expectedTime: undefined,
		// 		},
		// 		{
		// 			name: 'from undefined props(string type)',
		// 			initialDate: '2019-09-30 12:00',
		// 			newDate: undefined,
		// 			expectedDate: undefined,
		// 			expectedTime: undefined,
		// 		},
		// 		{
		// 			name: 'from props invalid date',
		// 			initialDate: new Date(),
		// 			newDate: new Date(''), // invalid date
		// 			expectedDate: undefined,
		// 			expectedTime: undefined,
		// 		},
		// 		{
		// 			name: 'from props valid date',
		// 			initialDate: new Date(),
		// 			newDate: new Date(2015, 3, 4, 12, 36),
		// 			expectedDate: new Date(2015, 3, 4),
		// 			expectedTime: { hours: '12', minutes: '36', seconds: '00' },
		// 		},
		// 		{
		// 			name: 'from props valid date with seconds',
		// 			initialDate: new Date(),
		// 			newDate: new Date(2015, 3, 4, 12, 36, 30),
		// 			expectedDate: new Date(2015, 3, 4),
		// 			expectedTime: { hours: '12', minutes: '36', seconds: '30' },
		// 			useSeconds: true,
		// 		},
		// 	],
		// );

		// cases(
		// 	'props update should NOT update state',
		// 	({ initialDate, newDate }) => {
		// 		// given
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} value={initialDate}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);
		// 		const previousState = wrapper.find('DateRangeConsumerDiv').props();

		// 		// when
		// 		act(() => {
		// 			wrapper.setProps({
		// 				value: newDate,
		// 			});
		// 		});
		// 		wrapper.update();

		// 		// then
		// 		const nextState = wrapper.find('DateRangeConsumerDiv').props();
		// 		expect(previousState.date).toEqual(nextState.date);
		// 		expect(previousState.time).toEqual(nextState.time);
		// 	},
		// 	[
		// 		{
		// 			name: "when date ref doesn't change",
		// 			initialDate: undefined,
		// 			newDate: undefined,
		// 		},
		// 		{
		// 			name: 'when the dates are equals',
		// 			initialDate: new Date(2015, 1, 5, 21, 52),
		// 			newDate: new Date(2015, 1, 5, 21, 52),
		// 		},
		// 	],
		// );
		// describe('on change', () => {
		// 	cases(
		// 		'should update state when date change',
		// 		({ expectedDate, expectedTime, textInput, dateFormat, useSeconds }) => {
		// 			// given
		// 			const event = { target: { value: textInput } };
		// 			const wrapper = mount(
		// 				<Manager id={DEFAULT_ID} dateFormat={dateFormat} useSeconds={useSeconds}>
		// 					<DateRangeConsumer />
		// 				</Manager>,
		// 			);

		// 			// when
		// 			act(() => {
		// 				wrapper
		// 					.find('DateRangeConsumerDiv')
		// 					.props()
		// 					.onDateChange(event, { date: expectedDate, textInput, errors: [] });
		// 			});
		// 			wrapper.update();

		// 			// then
		// 			const props = wrapper.find('DateRangeConsumerDiv').props();

		// 			expect(props.date).toEqual(expectedDate);
		// 			expect(props.time).toEqual(expectedTime);
		// 		},
		// 		[
		// 			{
		// 				name: 'with valid date',
		// 				textInput: '2015-01-15',
		// 				expectedDate: new Date(2015, 0, 15),
		// 				expectedTime: undefined,
		// 			},
		// 			{
		// 				name: 'with invalid date',
		// 				textInput: '2015aze-01-15',
		// 				expectedDate: '2015aze-01-15',
		// 				expectedTime: undefined,
		// 			},
		// 			{
		// 				name: 'with empty string',
		// 				textInput: '',
		// 				expectedDate: '',
		// 				expectedTime: undefined,
		// 			},
		// 			{
		// 				name: 'with custom date format',
		// 				textInput: '15/01/2015',
		// 				expectedDate: new Date(2015, 0, 15),
		// 				expectedTime: undefined,
		// 				dateFormat: 'DD/MM/YYYY',
		// 			},
		// 		],
		// 	);
		// 	cases(
		// 		'should update state when time change',
		// 		({ expectedDateRange, expectedDate, expectedTime, textInput, dateFormat, useSeconds }) => {
		// 			// given
		// 			const event = { target: { value: textInput } };
		// 			const wrapper = mount(
		// 				<Manager
		// 					id={DEFAULT_ID}
		// 					dateFormat={dateFormat}
		// 					useSeconds={useSeconds}
		// 					value={expectedDateRange}
		// 				>
		// 					<DateRangeConsumer />
		// 				</Manager>,
		// 			);

		// 			// when
		// 			act(() => {
		// 				wrapper
		// 					.find('DateRangeConsumerDiv')
		// 					.props()
		// 					.onTimeChange(event, { time: expectedTime, textInput, errors: [] });
		// 			});
		// 			wrapper.update();

		// 			// then
		// 			const props = wrapper.find('DateRangeConsumerDiv').props();

		// 			expect(props.date).toEqual(expectedDate);
		// 			expect(props.time).toEqual(expectedTime);
		// 		},
		// 		[
		// 			{
		// 				name: 'with valid time with seconds',
		// 				textInput: '15:45:22',
		// 				expectedDateRange: new Date(2015, 0, 15, 15, 45, 22),
		// 				expectedDate: new Date(2015, 0, 15),
		// 				expectedTime: { hours: '15', minutes: '45', seconds: '22' },
		// 				useSeconds: true,
		// 			},
		// 			{
		// 				name: 'with invalid time',
		// 				textInput: '15aze:45',
		// 				expectedDate: new Date(2015, 0, 15),
		// 				expectedDateRange: new Date(2015, 0, 15),
		// 				expectedTime: { hours: '15aze', minutes: '45', seconds: '00' },
		// 			},
		// 		],
		// 	);

		// 	it('should trigger props.onChange when date change', () => {
		// 		// given
		// 		const onChange = jest.fn();
		// 		const event = { target: { value: '2015-01-15' } };
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} onChange={onChange}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);
		// 		expect(onChange).not.toBeCalled();

		// 		// when
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onDateChange(event, {
		// 					date: new Date(2015, 0, 15),
		// 					textInput: '2015-01-15',
		// 					errors: [],
		// 				});
		// 		});
		// 		wrapper.update();

		// 		// then
		// 		expect(onChange).toBeCalled();
		// 		const args = onChange.mock.calls[0];
		// 		expect(args[0]).toBe(event);
		// 		expect(isNaN(args[1].datetime.getTime())).toBe(true);
		// 		expect(args[1].textInput).toBe('2015-01-15');
		// 		expect(args[1].errors).toEqual([
		// 			{ code: 'INVALID_TIME_EMPTY', message: 'Time is required' },
		// 		]);
		// 		expect(args[1].errorMessage).toBe('Time is required');
		// 	});

		// 	it('should trigger props.onChange with invalid date', () => {
		// 		// given
		// 		const onChange = jest.fn();
		// 		const event = { target: { value: '2015aze-01-15' } };
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} onChange={onChange}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);
		// 		expect(onChange).not.toBeCalled();

		// 		// when
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onDateChange(event, {
		// 					date: null,
		// 					textInput: '2015aze-01-15',
		// 					errors: [{ code: 'INVALID_DATE_FORMAT', message: 'Date format is invalid' }],
		// 				});
		// 		});
		// 		wrapper.update();

		// 		// then
		// 		expect(onChange).toBeCalled();
		// 		const args = onChange.mock.calls[0];
		// 		expect(args[0]).toBe(event);
		// 		expect(args[1].errorMessage).toBe('Date format is invalid');
		// 		expect(args[1].errors).toEqual([
		// 			{ code: 'INVALID_DATE_FORMAT', message: 'Date format is invalid' },
		// 		]);
		// 		expect(isNaN(args[1].datetime.getTime())).toBe(true);
		// 	});

		// 	it('should trigger props.onChange when time change', () => {
		// 		// given
		// 		const onChange = jest.fn();
		// 		const dateEvent = { target: { value: '2015-01-15' } };
		// 		const timeEvent = { target: { value: '12:45' } };
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} onChange={onChange}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);
		// 		expect(onChange).not.toBeCalled();

		// 		// when
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onDateChange(dateEvent, {
		// 					date: new Date(2015, 0, 15),
		// 					textInput: '2015-01-15',
		// 					errors: [],
		// 				});
		// 		});
		// 		wrapper.update();
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onTimeChange(timeEvent, {
		// 					time: { hours: '12', minutes: '45', seconds: '00' },
		// 					textInput: '12:45',
		// 					errors: [],
		// 				});
		// 		});
		// 		wrapper.update();
		// 		expect(onChange).toHaveBeenCalledTimes(2);
		// 		const args = onChange.mock.calls[1];
		// 		expect(args[0]).toBe(timeEvent);
		// 		expect(args[1].datetime).toEqual(new Date(2015, 0, 15, 12, 45));
		// 		expect(args[1].textInput).toBe('2015-01-15 12:45');
		// 		expect(args[1].errors).toEqual([]);
		// 		expect(args[1].errorMessage).toBe(null);
		// 	});
		// 	it('should trigger props.onChange with invalid time', () => {
		// 		// given
		// 		const onChange = jest.fn();
		// 		const dateEvent = { target: { value: '2015-01-15' } };
		// 		const timeEvent = { target: { value: '12dfd:45' } };
		// 		const wrapper = mount(
		// 			<Manager id={DEFAULT_ID} onChange={onChange}>
		// 				<DateRangeConsumer />
		// 			</Manager>,
		// 		);
		// 		expect(onChange).not.toBeCalled();

		// 		// when
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onDateChange(dateEvent, {
		// 					date: new Date(2015, 0, 15),
		// 					textInput: '2015-01-15',
		// 					errors: [],
		// 				});
		// 		});
		// 		wrapper.update();
		// 		act(() => {
		// 			wrapper
		// 				.find('DateRangeConsumerDiv')
		// 				.props()
		// 				.onTimeChange(timeEvent, {
		// 					textInput: '12dfd:45',
		// 					errors: [{ code: 'TIME_FORMAT_INVALID', message: 'Time is invalid' }],
		// 				});
		// 		});
		// 		wrapper.update();
		// 		expect(onChange).toHaveBeenCalledTimes(2);
		// 		const args = onChange.mock.calls[1];
		// 		expect(args[0]).toBe(timeEvent);
		// 		expect(isNaN(args[1].datetime.getTime())).toBe(true);
		// 		expect(args[1].textInput).toBe('2015-01-15 12dfd:45');
		// 		expect(args[1].errors).toEqual([
		// 			{ code: 'TIME_FORMAT_INVALID', message: 'Time is invalid' },
		// 		]);
		// 		expect(args[1].errorMessage).toBe('Time is invalid');
		// 	});
		// });
	});
	describe('picker change', () => {
		cases(
			'should update input',
			({ date, expectedTextInput, dateFormat }) => {
				// given
				const event = { preventDefault: () => {} };
				const wrapper = mount(
					<Manager id={DEFAULT_ID} dateFormat={dateFormat}>
						<DateRangeConsumer />
					</Manager>,
				);

				// when
				act(() => {
					wrapper
						.find('DateRangeConsumerDiv')
						.prop('inputManagement')
						.onFocus(event, 'startDate');
				});
				wrapper.update();

				act(() => {
					wrapper
						.find('DateRangeConsumerDiv')
						.prop('pickerManagement')
						.onSubmit(event, { date });
				});
				wrapper.update();

				// then
				const dateValue = wrapper.find('DateRangeConsumerDiv').props();
				expect(dateValue.startDate.textInput).toBe(expectedTextInput);
				expect(dateValue.startDate.value).toEqual(date);
			},
			[
				{
					name: 'with valid date',
					date: new Date(2015, 0, 15),
					expectedTextInput: '2015-01-15',
				},
				{
					name: 'with custom date format',
					date: new Date(2015, 0, 15),
					dateFormat: 'DD/MM/YYYY',
					expectedTextInput: '15/01/2015',
				},
			],
		);

		it('should trigger props.onChange with valid date', () => {
			// given
			const onChange = jest.fn();
			const event = { target: {}, preventDefault: () => {} };
			const wrapper = mount(
				<Manager id={DEFAULT_ID} onChange={onChange}>
					<DateRangeConsumer />
				</Manager>,
			);

			expect(onChange).not.toBeCalled();

			// when
			act(() => {
				wrapper
					.find('DateRangeConsumerDiv')
					.prop('inputManagement')
					.onFocus(event, 'startDate');
			});
			wrapper.update();
			act(() => {
				wrapper
					.find('DateRangeConsumerDiv')
					.prop('pickerManagement')
					.onSubmit(event, {
						date: new Date(2015, 0, 15),
					});
			});
			wrapper.update();
			// then
			expect(onChange).toBeCalledWith(
				event,
				{
					startDate: new Date(2015, 0, 15),
					endDate: undefined,
					errorMessage: undefined,
					errors: [],
					field: 'startDate',
					origin: 'RANGE_PICKER',
				},
				'endDate',
			);
		});
	});
});
