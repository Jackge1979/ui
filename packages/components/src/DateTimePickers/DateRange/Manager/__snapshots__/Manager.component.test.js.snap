// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange.Manager should render its children 1`] = `
<Context.Provider
  value={
    Object {
      "endDate": Object {
        "textInput": "2017-04-10",
        "value": 2017-04-09T22:00:00.000Z,
      },
      "endTime": Object {
        "textInput": "00:00",
        "value": undefined,
      },
      "inputManagement": Object {
        "onEndChange": [Function],
        "onStartChange": [Function],
        "onTimeChange": [Function],
        "placeholder": "YYYY-MM-DD",
        "timePlaceholder": "HH:mm",
      },
      "pickerManagement": Object {
        "onEndChange": [Function],
        "onStartChange": [Function],
        "onTimeChange": [Function],
      },
      "startDate": Object {
        "textInput": "2017-04-04",
        "value": 2017-04-03T22:00:00.000Z,
      },
      "startTime": Object {
        "textInput": "00:00",
        "value": undefined,
      },
    }
  }
>
  <DateRangeConsumer />
</Context.Provider>
`;
